### PR TITLE
Workaround webkit disrespecting e.preventDefault()

### PR DESCRIPTION
**Pull request by anandthakker**
Follow-up to #6248 to work around https://bugs.webkit.org/show_bug.cgi?id=182521

### CODE REVIEW

**Comment by jfirebaugh** [comment]
Let's just bind both `touchstart` and `touchmove` aggressively everywhere. I'm done with passive event listers.

File: src/ui/bind_handlers.js [hunk 1]
```diff
@@ -38,8 +42,13 @@ export default function bindHandlers(map: Map, options: {}) {
...
     DOM.addEventListener(el, 'mousemove', onMouseMove);
     DOM.addEventListener(el, 'mouseover', onMouseOver);
-    DOM.addEventListener(el, 'touchstart', onTouchStart, {passive: false});
-    DOM.addEventListener(el, 'touchmove', onTouchMove, {passive: true}); // `passive: true` because onTouchMove only sends a map event.
+    // Bind touchstart with passive: true because onTouchStart only fires a map event
+    DOM.addEventListener(el, 'touchstart', onTouchStart, {passive: true});
+    // Bind touchmove with passive: false on iOS because, even though
+    // onTouchMove only fires a map event, binding with passive: true causes
+    // iOS not to respect e.preventDefault() in _other_ handlers, even if they
+    // are non-passive.  https://bugs.webkit.org/show_bug.cgi?id=182521
```
Comment (jfirebaugh) [inline src/ui/bind_handlers.js]: Is this the right WebKit bug URL?

Comment (anandthakker) [reply src/ui/bind_handlers.js]: Whoops, I must have copied from the wrong tab, thanks.

**Comment by anandthakker** [comment]
@jfirebaugh updated

File: src/ui/bind_handlers.js [hunk 2]
```diff
@@ -38,8 +38,15 @@ export default function bindHandlers(map: Map, options: {}) {
...
+    // Bind touchstart and touchmove with passive: false because, even though
+    // they only fire a map events and therefore could theoretically be
+    // passive, binding with passive: true causes iOS not to respect
+    // e.preventDefault() in _other_ handlers, even if they are non-passive
+    // (see https://bugs.webkit.org/show_bug.cgi?id=184251)
     DOM.addEventListener(el, 'touchstart', onTouchStart, {passive: false});
-    DOM.addEventListener(el, 'touchmove', onTouchMove, {passive: true}); // `passive: true` because onTouchMove only sends a map event.
+    DOM.addEventListener(el, 'touchmove', onTouchMove, {passive: false});
+
     DOM.addEventListener(el, 'touchend', onTouchEnd);                    // The real action is in DragPanHandler and TouchZoomRotateHandler.
```
Comment (jfirebaugh) [inline src/ui/bind_handlers.js]: You can remove the comment here.

**Pull request merged**